File: aoj/volume100/php/A10016.php
<?php
function grade($m, $f, $r) {
    if ($m == -1 || $f == -1)
        return "F";
    else if ($m + $f >= 80)
        return "A";
    else if ($m + $f >= 65)
        return "B";
    else if ($m + $f >= 50)
        return "C";
    else if ($m + $f >= 30 && $r >= 50)
        return "C";
    else if ($m + $f >= 30)
        return "D";
    else
        return "F";
}

while (true) {
    fscanf(STDIN, "%d %d %d", $m, $f, $r);
    if ($m == -1 && $f == -1 && $r == -1)
        break;

    printf("%s\n", grade($m, $f, $r));
}

File: aoj/volume100/python3/A10016.py
import sys

def grade(m, f, r):
    if m == -1 or f == -1:
        return 'F'
    elif m+f >= 80:
        return 'A'
    elif m+f >= 65:
        return 'B'
    elif m+f >= 50:
        return 'C'
    elif m+f >= 30 and r >= 50:
        return 'C'
    elif m+f >= 30:
        return 'D'
    elif m+f < 30:
        return 'F'

for line in sys.stdin:
    m, f, r = map(int, line.rstrip().split())
    if m == -1 and f == -1 and r == -1:
        break

    print(grade(m, f, r))
